imx-core-sdk-kotlin-jvm/src/main/kotlin/com/immutable/sdk/crypto/StarkKey.kt

```diff
@@ -59,6 +103,43 @@ object StarkKey {
         return key.mod(ORDER).toString(Constants.HEX_RADIX)
     }
 
+    /**
+     * Generates the Stark key pair from the [seed] and [path]
+     */
+    private fun getKeyPairFromPath(seed: String, path: String): String {
+        val master =
+            Bip32ECKeyPair.generateKeyPair(BigInteger(seed, Constants.HEX_RADIX).toByteArray())
+        val pathArray =
+            path.split("/") // Example: "m/2645'/579218131'/211006541'/9971226311'/947333111'/1"
+        val p = pathArray
+            .subList(1, pathArray.size) // Ignore the first one
+            .map {
+                val isHardened = it.endsWith("'")
+                it
+                    .replace("'", "") // Remove hardened notation
+                    .toInt() // Convert string to int
+                    .run {
+                        if (isHardened) this or HARDENED_BIT else this
+                    }
+            }.toIntArray()
+        val gk = grindKey(
+            Bip32ECKeyPair.deriveKeyPair(master, p)
+                .privateKeyBytes33.drop(1).toHexString()
+        )
+        return gk.sanitizeBytes()
+    }
+
+    /**
+     * Generates a deterministic Stark private key from the provided signer.
+     * @returns the private key as a hex string
+     */
+    @Suppress("MagicNumber")
+    fun generateLegacyKeyPair(seed: String, ethereumAddress: String): String {
```

Comment:
should we do `generateLegacyPrivateKey` then since you're not returning the key pair?

Reply:
Changed the name to `generateLegacyStarkPrivateKey` to match TS Core SDK.